packages/valory/skills/registration_abci/behaviours.py

```diff
@@ -67,8 +122,277 @@ def async_act(self) -> Generator:
+            self.context.outbox.put_message(message=message, context=context)
+
+        if all(self.registered_addresses.values()):
+            log_message = self.LogMessages.collection_complete
+            self.context.logger.info(f"{log_message}: {self.registered_addresses}")
+            self.collection_complete = True
+        return self.collection_complete
+
+    def request_update(self) -> Generator[None, None, bool]:
+        """Make HTTP POST request to update agent's local Tendermint node"""
+
+        url = self.tendermint_parameter_url
+        genesis_data = format_genesis_data(self.registered_addresses)
+        log_message = self.LogMessages.request_update
+        self.context.logger.info(f"{log_message}: {genesis_data}")
+
+        content = json.dumps(genesis_data).encode(self.ENCODING)
+        result = yield from self.get_http_response(
+            method="POST", url=url, content=content
+        )
+        response = self._decode_result(result, self.LogMessages.failed_update)
+        if response is None:
+            return False
+
+        log_message = self.LogMessages.response_update
+        self.context.logger.info(f"{log_message}: {response}")
+        self.updated_genesis_data.update(genesis_data)
+        return True
+
+    def async_act(self) -> Generator:
+        """Act asynchronously"""
+
+        if not self.params.share_tm_config_on_startup:
+            yield from super().async_act()
+
+        self.context.logger.info(f"My address: {self.context.agent_address}")
+
+        # collect personal Tendermint configuration
+        if not self.local_tendermint_params:
+            successful = yield from self.get_tendermint_configuration()
+            if not successful:
+                yield from self.sleep(self.params.sleep_time)
+                return
+
+        # make service registry contract call
+        if not self.registered_addresses:
+            successful = yield from self.get_addresses()
+            if not successful:
+                yield from self.sleep(self.params.sleep_time)
+                return
+
+        # collect Tendermint config information from other agents
+        if not self.collection_complete:
+            successful = self.request_tendermint_info()
+            if not successful:
+                yield from self.sleep(self.params.sleep_time)
+                return
+
+        # update Tendermint configuration
+        if not self.updated_genesis_data:
+            successful = yield from self.request_update()
+            if not successful:
+                yield from self.sleep(self.params.sleep_time)
+                return
+
+        # restart Tendermint with updated configuration
+        successful = yield from self.reset_tendermint_with_wait(on_startup=True)
+        if not successful:
+            yield from self.sleep(self.params.sleep_time)
+            return
+
+        yield from super().async_act()
```

Comment:
Since we're running this at the start as well

```python
        if not self.params.share_tm_config_on_startup:
             yield from super().async_act()
``` 

Why not make the execution flow like this

```python
if self.params.share_tm_config_on_startup:
    # collect tendermint config

yield from super().async_act()
```

Reply:
https://github.com/valory-xyz/open-autonomy/pull/909#discussion_r895545012